app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +23,35 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
+    case RECEIVE_USERS: {
+      // Get the sorting key from the Redux store
+      let sort_key = state.sort.key;
+
+       // Initialize a variable to hold the user list. If there are existing users in the state,
+      //  use that list; otherwise, use the user list from the action data.
+      let user_list = state.users.length ? state.users : action.data.course.users;
```

Comment:
Are you sure this is correct? Why would you want to read the users from the state when we are receiving users?